spec/ValidationAndPasswordsReset.spec.js

```diff
@@ -1082,4 +1082,56 @@ describe('Custom Pages, Email Verification, Password Reset', () => {
         done();
       });
   });
+
+  it('should resolve on an invalid reset password with default server parameters', async () => {
+    await reconfigureServer({
+      appName: 'coolapp',
+      publicServerURL: 'http://localhost:1337/1',
+      emailAdapter: MockEmailAdapterWithOptions({
+        fromAddress: 'parse@example.com',
+        apiKey: 'k',
+        domain: 'd',
+      }),
+    });
+    const result = await Parse.User.requestPasswordReset('test@example.com');
+    expect(result).toEqual({});
+  });
+
+  it('validate resetPasswordSuccessonInvalidEmail', async () => {
+    await expectAsync(
+      reconfigureServer({
+        appName: 'coolapp',
+        publicServerURL: 'http://localhost:1337/1',
+        emailAdapter: MockEmailAdapterWithOptions({
+          fromAddress: 'parse@example.com',
+          apiKey: 'k',
+          domain: 'd',
+        }),
+        passwordPolicy: {
+          resetPasswordSuccessOnInvalidEmail: [],
+        },
+      })
+    ).toBeRejectedWith('resetPasswordSuccessOnInvalidEmail must be a boolean value');
+  });
+
+  it('should throw on an invalid reset password with resetPasswordSuccessOnInvalidEmail', async () => {
+    await reconfigureServer({
+      appName: 'coolapp',
+      publicServerURL: 'http://localhost:1337/1',
+      emailAdapter: MockEmailAdapterWithOptions({
+        fromAddress: 'parse@example.com',
+        apiKey: 'k',
+        domain: 'd',
+      }),
+      passwordPolicy: {
+        resetPasswordSuccessOnInvalidEmail: false,
+      },
+    });
+    await expectAsync(Parse.User.requestPasswordReset('test@example.com')).toBeRejectedWith(
+      new Parse.Error(
+        Parse.Error.OBJECT_NOT_FOUND,
+        'A user with the email test@example.com does not exist.'
```

Comment:
I don't think the email address should be contained in an error message. An email is PII.